aws_lambda_builders/workflows/custom_make/workflow.py

```diff
@@ -34,6 +34,21 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
         # Find the logical id of the function to be built.
         options = kwargs.get("options") or {}
         build_logical_id = options.get("build_logical_id", None)
+        working_directory = options.get("working_directory")
+
+        if working_directory is None:
+            working_directory = scratch_dir
+        else:
+            if self.os_utils.isabspath(working_directory):
+                # Working directory should be relative to the source_dir, so when we copy the source_dir to scratch_dir
+                # the working directory value is still valid
+                working_directory = self.os_utils.relpath(working_directory, source_dir)
+
+            # join the working directory to the scratch_dir
+            working_directory = self.os_utils.normpath(self.os_utils.join_relpath(working_directory, scratch_dir))
```

Comment:
is working directory going to be a sub dir within scratch dir all the time? then is this more of a naming? its not really a directory that can exist by itself.

Reply:
it should be always a sub directory within source_code path, to be copied to the scratch_dir, and then we can run parallel build.

Reply:
removed